src/overlays/actors/ovl_En_Raf/z_en_raf.c

```diff
@@ -80,65 +178,787 @@ static DamageTable D_80A1939C = {
+            if ((limbIndex > CARNIVOROUS_LILY_PAD_LIMB_FLOWER) && (limbIndex < CARNIVOROUS_LILY_PAD_LIMB_ROOTS)) {
+                Matrix_RotateY((this->chewLimbRot[limbIndex].y * globalCtx->gameplayFrames), MTXMODE_APPLY);
+                Matrix_InsertXRotation_s((this->chewLimbRot[limbIndex].x * globalCtx->gameplayFrames), MTXMODE_APPLY);
+                Matrix_InsertZRotation_s((this->chewLimbRot[limbIndex].z * globalCtx->gameplayFrames), MTXMODE_APPLY);
+                Matrix_Scale(this->chewScale + 1.0f, 1.0f, 1.0f, MTXMODE_APPLY);
+                Matrix_InsertZRotation_s(-(this->chewLimbRot[limbIndex].z * globalCtx->gameplayFrames), MTXMODE_APPLY);
+                Matrix_InsertXRotation_s(-(this->chewLimbRot[limbIndex].x * globalCtx->gameplayFrames), MTXMODE_APPLY);
+                Matrix_RotateY(-(this->chewLimbRot[limbIndex].y * globalCtx->gameplayFrames), MTXMODE_APPLY);
+            }
+            break;
+
+        case EN_RAF_PETAL_SCALE_TYPE_IDLE_OR_THROW:
+            if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_MIDDLE_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_MIDDLE_SEGMENT)) {
+                for (i = 0; i < ARRAY_COUNT(sSpitAnimationCheckFrames); i++) {
+                    if ((s16)this->skelAnime.curFrame == sSpitAnimationCheckFrames[i]) {
+                        Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sMiddleSegmentTargetScaleDuringSpit[i]);
+                    }
+                }
+            }
+
+            if ((limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_1_UPPER_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_3_UPPER_SEGMENT) ||
+                (limbIndex == CARNIVOROUS_LILY_PAD_LIMB_TRAP_2_UPPER_SEGMENT)) {
+                for (i = 0; i < 4; i++) {
+                    if ((s16)this->skelAnime.curFrame == sSpitAnimationCheckFrames[i]) {
+                        Math_Vec3f_Copy(&this->targetLimbScale[limbIndex], &sUpperSegmentTargetScaleDuringSpit[i]);
+                    }
+                }
+            }
+            break;
+    }
+
+    Matrix_Scale(this->limbScale[limbIndex].x, this->limbScale[limbIndex].y, this->limbScale[limbIndex].z,
+                 MTXMODE_APPLY);
+}
+
+void EnRaf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnRaf* this = THIS;
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_8012C2DC(globalCtx->state.gfxCtx);
+    SkelAnime_DrawTransformFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.jointTable,
+                                   this->skelAnime.dListCount, NULL, NULL, EnRaf_TransformLimbDraw, &this->dyna.actor);
+
+    if (this->action == EN_RAF_ACTION_EXPLODE) {
+        EnRaf_DrawParticles(this, globalCtx);
+    }
+}
+
+void EnRaf_InitializeParticle(EnRaf* this, Vec3f* position, Vec3f* velocity, Vec3f* acceleration, f32 scale,
```

Comment:
I think OoT has moved away from "particle" to just calling these "special effects" or something, would recommend investigating the current terminology to keep us in sync. (I don't have a strong opinion on which is better, so happy to go the OoT way.)

Reply:
Hmmm, well, most of the bosses call them "effects" because they use `globalCtx->specialEffects`. I can find one instance of an actor in OoT using "effect" *without* using `globalCtx->specialEffects`, which is En_G_Switch. Meanwhile, there are a few more instances of "particle" being used for this purpose instead, including for actors whose code for handling these things looks nearly-identical to Raf's. Unfortunately, there isn't a ton of consistency in OoT either. My gut says to leave this for now and we can decide on the particle vs effect nomenclature later, but what do you think?

Reply:
Yeah, let's not worry about it for now then... I did think there'd been a push for consensus on effects. 🤔

Reply:
I did a quick discord search for "particle" and I didn't find anything recently. So if there *was* a consensus, maybe it was a while ago?